specification/webpubsub/data-plane/WebPubSub/stable/2021-10-01/webpubsub.json

```diff
@@ -0,0 +1,1173 @@
+            "type": "string",
+            "pattern": "^[A-Za-z][A-Za-z0-9_`,.[\\]]{0,127}$"
+          },
+          {
+            "in": "query",
+            "name": "userId",
+            "description": "User Id.",
+            "type": "string"
+          },
+          {
+            "in": "query",
+            "name": "role",
+            "description": "Roles that the connection with the generated token will have.",
+            "type": "array",
+            "items": {
+              "type": "string"
+            },
+            "collectionFormat": "multi"
+          },
+          {
+            "in": "query",
+            "name": "minutesToExpire",
+            "description": "The expire time of the generated token.",
+            "type": "integer",
+            "format": "int32",
+            "default": 60
+          },
+          {
+            "in": "query",
+            "name": "api-version",
+            "description": "The version of the REST APIs.",
+            "type": "string",
+            "default": "2021-10-01"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/ClientTokenResponse"
+            }
+          },
+          "default": {
+            "description": "Error response",
+            "x-ms-error-response": true
+          }
+        }
+      }
+    },
+    "/api/hubs/{hub}/:closeConnections": {
+      "post": {
+        "tags": [
+          "webpubsub"
+        ],
+        "summary": "Close the connections in the hub.",
+        "operationId": "WebPubSub_CloseAllConnections",
+        "parameters": [
+          {
+            "in": "path",
+            "name": "hub",
+            "description": "Target hub name, which should start with alphabetic characters and only contain alpha-numeric characters or underscore.",
+            "required": true,
+            "type": "string",
+            "pattern": "^[A-Za-z][A-Za-z0-9_`,.[\\]]{0,127}$"
+          },
+          {
+            "in": "query",
+            "name": "excluded",
+            "description": "Exclude these connectionIds when closing the connections in the hub.",
+            "type": "array",
+            "items": {
+              "type": "string"
+            },
+            "collectionFormat": "multi"
+          },
+          {
+            "in": "query",
+            "name": "reason",
+            "description": "The reason closing the client connection.",
+            "type": "string"
+          },
+          {
+            "in": "query",
+            "name": "api-version",
+            "description": "The version of the REST APIs.",
+            "type": "string",
+            "default": "2021-10-01"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success"
+          },
+          "default": {
+            "description": "Error response",
+            "x-ms-error-response": true
```

Comment:
Thanks, looks cool, will follow

Reply:
Question: there could be error responses returned **earlier** when a client calls the REST API, for example, when the load balancer returns 502, in such case, the response body would probably be some plain-text body. Will that break the client if the `default` response only defines the `ErrorDetail` response?

Reply:
@vicancy - If I'm understanding this correctly, this would be an error that happens BEFORE the service is involved--e.g. the request never gets to the service. If this is accurate, then we should not worry about modeling this in the swagger document. We should focus on the errors that the service throws and work to be consistent there. I'll also check with our SDK team that our libraries are resilient in this scenario.

Reply:
Thanks for the effort Mark. Yeah my only concern is if our SDK clients can handle this without error. Anyway, updated the swagger.

Reply:
Great! Thank you!